lib/Controller/SAMLController.php

```diff
@@ -319,6 +319,13 @@ public function assertionConsumerService() {
 	public function singleLogoutService() {
 		$isFromGS = ($this->config->getSystemValue('gs.enabled', false) &&
 					 $this->config->getSystemValue('gss.mode', '') === 'master');
+
+		// Some IDPs send the SLO request via POST, but OneLogin php-saml only handles GET.
+		// To hack around this issue we copy the request from _POST to _GET.
+		if(!empty($_POST['SAMLRequest'])) {
```

Comment:
use `$this->request->getParam(...)` instead. it's an abstraction of the `$_GET` and `$_POST` vars that allows unit testing

Reply:
The problem is that I need to check for $_POST here explicitly to copy the value to $_GET for the underlying php-saml lib. If I would check via `getParam()`, I wouldn't know whether the data was passed via GET or POST and I wouldn't know if I need to fill $_GET.

The only thing I can imagine would be to set $_GET unconditionally like:
```
$_GET['SAMLRequest'] = $this->request->getParam('SAMLRequest');
```

Is that what you mean?

Reply:
I'm sorry, I didn't know about the internal handling of superglobals in php-saml. In that case I guess it's fine to access the globals directly.
It is possible to access only get or post vars via IRequest::get($name) and IRequest::post($name) but it will throw an exception if you access get on post requests and vice versa.

Reply:
@ChristophWurst so leave it like that?

Reply:
yes!

Reply:
Review? :D

Reply:
Yes, someone who knows the SAML code should have a look :)

Reply:
https://github.com/nextcloud/user_saml/pull/340#event-2423557266 :see_no_evil: